src/Components/AllArticles.jsx

```diff
@@ -38,6 +39,28 @@ const AllArticles = () => {
     );
   };
 
+  const handleSortChange = (e) => {
+    const sortKey = e.target.value;
+    setSortBy(sortKey);
+    const sortedArticles = [...articles];
+
+    if (sortKey === "byDate") {
+      sortedArticles.sort(
+        (a, b) => new Date(b.created_at) - new Date(a.created_at)
+      );
+    } else if (sortKey === "commentCount") {
+      sortedArticles.sort((a, b) => b.comment_count - a.comment_count);
+    } else if (sortKey === "votes") {
+      sortedArticles.sort((a, b) => b.votes - a.votes);
+    } else if (sortKey === "ASC") {
```

Comment:
with ASC & DESC I would think about having them separate as they should be able to effect each of the choices . eg votes ASC or DESC